build/ci/official.yml

```diff
@@ -18,37 +21,61 @@ variables:
   TeamName: DotNet-Project-System
   BuildPlatform: any cpu
   DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
-  
+
+###################################################################################################################################################################
+# INSTALLATION
+###################################################################################################################################################################
+
 steps:
-- task: ms-vseng.MicroBuildTasks.30666190-6959-11e5-9f96-f56098202fef.MicroBuildSigningPlugin@1
+- task: MicroBuildSigningPlugin@4
   displayName: Install Signing Plugin
   inputs:
     signType: real
     esrpSigning: true
-    
-- task: ms-vseng.MicroBuildTasks.32f78468-e895-4f47-962c-58a699361df8.MicroBuildSwixPlugin@1
+
+# This is use as a PackageReference with ExcludeAssets="all", which means it is not executed.
+# RepoToolset has custom logic to run the contents of this plugin, so it is not executed normally.
+- task: MicroBuildSwixPlugin@1
   displayName: Install Swix Plugin
 
-- task: NuGetCommand@2
-  displayName: Restore 'Microsoft.DotNet.IBCMerge' package
+# This requires the OptimizationInputs via the 'Publish OptimizationInputs drop' step in the release pipeline.
+# To generate ProfilingInputs for the first time, set this input: ShouldSkipOptimize: true
+# This plugin will restore MicroBuild.Plugins.OptProf which is required to create ProfilingInputs data.
+# The projects include OptProf.targets which uses MicroBuild.Plugins.OptProf to run the target GenerateProfilingInputs on each project.
+- task: MicroBuildOptProfPlugin@6
+  displayName: Install OptProf Plugin
   inputs:
-    command: restore
-    feedsToUse: config
-    restoreSolution: 'build\proj\internal\RestoreIBCMerge.csproj'
-    nugetConfigPath: 'NuGet.config'
+    ProfilingInputsDropName: 'ProfilingInputs/$(System.TeamProject)/$(Build.Repository.Name)/$(Build.SourceBranchName)/$(Build.BuildId)'
+    OptimizationInputsLookupMethod: 'DropPrefix'
+    DropNamePrefix: 'OptimizationInputs/$(System.TeamProject)/$(Build.Repository.Name)'
+    AccessToken: '$(System.AccessToken)'
+    feedSource: 'https://devdiv.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+    ShouldSkipOptimize: false
+    NumberCommitsToSearch: '100'
+
+###################################################################################################################################################################
+# BUILD SOLUTION
+###################################################################################################################################################################
 
-# TODO: Merged PIAs - reset to /ibc once private feeds are removed and Microsoft.DotNet.IBCMerge is reset
```

Comment:
Does IBC still play a role in our builds at all? Can it be removed from `build.cmd` and elsewhere?

Reply:
IBC is simply the tool used to merge optimization data into an assembly. The flags in `build.cmd` are quite out-of-date but I didn't want to go through and change the calls to `build.cmd` that assume those flags are there. Instead, I've noted that I want to clean up and simplify the build hierarchy, which would include removing unused flags, as a separate issue I'll make. I've removed IBC usage/references from everywhere else except `build.cmd` in this PR. Note: OptProf itself uses IBCMerge.exe but that is part of the plugin's responsibilities. So we don't need to do anything manually with it.

Reply:
Created an issue for this here: https://github.com/dotnet/project-system/issues/7868